http-server-netty/src/test/groovy/io/micronaut/http/server/netty/cors/CorsFilterSpec.groovy

```diff
@@ -112,198 +133,251 @@ class CorsFilterSpec extends Specification {
-        result.get().status == HttpStatus.FORBIDDEN
+
+        when:
+        MutableHttpResponse<?> response = result.get()
+
+        then:
+        HttpStatus.FORBIDDEN == response.status()
+        response.headers.names().isEmpty()
     }
 
-    void "test handleRequest with disallowed header (not preflight)"() {
+    void "with disallowed header (not preflight) the request is passed through because allowed headers are only checked for preflight requests"() {
         given:
-        HttpRequest request = Mock(HttpRequest)
-        HttpHeaders headers = Mock(HttpHeaders)
-        request.getHeaders() >> headers
+        String origin = 'http://www.foo.com'
+        HttpRequest request = createRequest(origin)
+        request.getMethod() >> HttpMethod.GET
 
-        def config = new HttpServerConfiguration.CorsConfiguration()
         CorsOriginConfiguration originConfig = new CorsOriginConfiguration()
         originConfig.allowedOrigins = ['http://www.foo.com']
         originConfig.allowedMethods = [HttpMethod.GET]
-        config.configurations = new LinkedHashMap<String, CorsOriginConfiguration>()
-        config.configurations.put('foo', originConfig)
+        HttpServerConfiguration.CorsConfiguration config = enabledCorsConfiguration([foo: originConfig])
         CorsFilter corsHandler = buildCorsHandler(config)
 
         when:
-        def result = corsHandler.handleRequest(request)
+        Optional<MutableHttpResponse<?>> result = Mono.from(corsHandler.doFilter(request, okChain())).blockOptional()
 
-        then: "the request is passed through because allowed headers are only checked for preflight requests"
-        2 * headers.getOrigin() >> Optional.of('http://www.foo.com')
-        1 * request.getMethod() >> HttpMethod.GET
-        !result.isPresent()
-        0 * headers.get(ACCESS_CONTROL_REQUEST_HEADERS, _)
+        then:
+        result.isPresent()
+
+        when:
+        MutableHttpResponse<?> response = result.get()
+
+        then:
+        HttpStatus.OK == response.status()
+        response.headers.names().size() == 3
+        response.headers.find { it.key == 'Access-Control-Allow-Origin' }
+        response.headers.find { it.key == 'Vary' }
+        response.headers.find { it.key == 'Access-Control-Allow-Credentials' }
+        response.headers.find { it.key == 'Access-Control-Allow-Origin' }.value == ['http://www.foo.com']
+        response.headers.find { it.key == 'Vary' }.value == ['Origin']
+        response.headers.find { it.key == 'Access-Control-Allow-Credentials' }.value == [StringUtils.TRUE]
     }
 
     void "test preflight handleRequest with disallowed header"() {
         given:
-        HttpRequest request = Mock(HttpRequest)
-        HttpHeaders headers = Mock(HttpHeaders)
-        request.getHeaders() >> headers
-        def config = new HttpServerConfiguration.CorsConfiguration()
+        String origin = 'http://www.foo.com'
+        HttpHeaders headers = Stub(HttpHeaders) {
```

Comment:
do we need a mock here, or is there an implementation we can use?

Reply:
In other places, we use `MockHttpHeaders` but I think in this test the Stub is ok.